larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewTransverseTracksAlgorithm.h

```diff
@@ -24,25 +29,69 @@ class TransverseMatrixTool;
+    /**
+     *  @brief  The overridden function called by the base class which instigates calculation of the two view overlap result
+     *
+     *  @param  pCluster1 the view 0 cluster
+     *  @param  pCluster2 the view 1 cluster
+     */
+    void CalculateOverlapResult(const pandora::Cluster *const pCluster1, const pandora::Cluster *const pCluster2, 
+        const pandora::Cluster *const);
+
+    /**
+     *  @brief  Calculates the two view overlap result
+     *
+     *  @param  pCluster1 the view 0 cluster
+     *  @param  pCluster2 the view 1 cluster
+     *  @param  overlapResult the two view overlap result
+     */
+    pandora::StatusCode CalculateOverlapResult(const pandora::Cluster *const pCluster1, const pandora::Cluster *const pCluster2, 
+        TwoViewTransverseOverlapResult &overlapResult);
+
+    /**
+     *  @brief  Calculates the fraction of the sliding windows that contains charge bins that locally match
+     *
+     *  @param  discreteProbabilityVector1 the view 0 discrete probability vector containing the charge information
+     *  @param  pCluster2 the view 1 discrete probability vector containing the charge information
+     *
+     *  @result the matched window fraction
+     */
+    float CalculateLocalMatchingFraction(const DiscreteProbabilityVector &discreteProbabilityVector1, 
+        const DiscreteProbabilityVector &discreteProbabilityVector2);
+
+    /**
+     *  @brief  Runs the list of matching tools on each element in the overlap matrix 
+     */
     void ExamineOverlapContainer();
+
+    /**
+     *  @brief  Reads the settings from the XML 
+     *
+     *  @param  xmlHandle the xml handle
+     */
     pandora::StatusCode ReadSettings(const pandora::TiXmlHandle xmlHandle);
 
     typedef std::vector<TransverseMatrixTool*> MatrixToolVector;
-    MatrixToolVector            m_algorithmToolVector;      ///< The algorithm tool vector
+    MatrixToolVector            m_algorithmToolVector;           ///< The algorithm tool vector
 
-    unsigned int                m_nMaxMatrixToolRepeats;    ///< The maximum number of repeat loops over matrix tools
+    unsigned int                m_nMaxMatrixToolRepeats;         ///< The maximum number of repeat loops over matrix tools
+    unsigned int                m_downsampleFactor;              ///< The downsampling (hit merging) applied to hits in the overlap region
+    unsigned int                m_minSamples;                    ///< The minimum number of samples needed for comparing charges
+    unsigned int                m_nPermutations;                 ///< The number of permutations for calculating p-values
+    float                       m_localMatchingScoreThreshold;   ///< The minimum score to classify a local region as matching
+    std::random_device          m_randomDevice;                  ///< The random device used for seeding the number generator
```

Comment:
Since this bit is relevant to me as well, I'll stick a comment here to see if its worth discussing.

Do we want to use `std::random_device` here, since it will give different results run to run? There is enough permutations being done that it realistically won't matter, but its worth discussing at least.

In my 3D RANSAC code, I'd provisionally removed it and gone for a hard-coded seed (well, hard-coded in that I use the length of the input vector, so it varies particle to particle, but is the same across all machines).

Reply:
I think this is a really good point.  I'd largely left this as a place holder for the random need.
I think it warrants a good discussion as well.
I like reproducibility in the long term, but coming up with the 'ideal' seed is one of those "don't let perfect be the enemy of good enough" moments so I opted for completely random for now.

In the long term, I think we probably want a well defined seed system, but it might need to extend beyond container size.  Some kind of hash based on a bunch of well defined variables are something.

Reply:
Yes, this is an excellent point. Use of random numbers, etc. is fine, but we need our jobs to be reproducible, and have spent a lot of time in the past checking that they are reproducible to a very high level (no known issues currently). Anything that ensures we can debug individual events safely and reproducibly(/portably) should just be fine to start off with, then can obviously increase level of sophistication if alg requires.

Reply:
I just ran a quick test, using the summed cluster hits as the seed vs using the random device--seeded rng, which produced near enough identical results, I guess because of how many permutations are involved in the permutation test (10,000).  I will, however, swap to using the summed cluster hits as the seed for now.

I think we'll need a much larger discussion about the randomness down the line.